wolox-books/src/app/store/books.reducer.ts

```diff
@@ -0,0 +1,19 @@
+import { Action } from '@ngrx/store'
+import { Book, ShoppingCarState } from '../models/lists-books.model'
+import * as ListBooksCarActions from './books.actions'
+
+const initialState: ShoppingCarState = { books: [], loading: false}
+
+export function reducer(state = initialState, action: ListBooksCarActions.Actions) {
+    switch(action.type) {
+        case ListBooksCarActions.ADD_BOOKS:
+            return {...state, books: [...state.books, action.payload]}
+        
+        case ListBooksCarActions.REMOVE_BOOKS:
+            state.books.splice(action.payload, 1)
```

Comment:
This works, but i think it's a better way if you do it like this: {...state, books: state.books.splice(action.payload,1) }